presto-geospatial/src/main/java/io/prestosql/plugin/geospatial/EncodedPolylineFunctions.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.geospatial;
+
+import com.esri.core.geometry.MultiPath;
+import com.esri.core.geometry.MultiVertexGeometry;
+import com.esri.core.geometry.Point;
+import com.esri.core.geometry.Polyline;
+import com.esri.core.geometry.ogc.OGCGeometry;
+import com.esri.core.geometry.ogc.OGCLineString;
+import com.google.common.base.Joiner;
+import io.airlift.slice.DynamicSliceOutput;
+import io.airlift.slice.Slice;
+import io.prestosql.geospatial.GeometryType;
+import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.function.Description;
+import io.prestosql.spi.function.ScalarFunction;
+import io.prestosql.spi.function.SqlType;
+import io.prestosql.spi.type.StandardTypes;
+
+import java.util.EnumSet;
+import java.util.Set;
+
+import static io.prestosql.geospatial.GeometryType.LINE_STRING;
+import static io.prestosql.geospatial.GeometryType.MULTI_POINT;
+import static io.prestosql.geospatial.serde.GeometrySerde.deserialize;
+import static io.prestosql.geospatial.serde.GeometrySerde.serialize;
+import static io.prestosql.plugin.geospatial.GeometryType.GEOMETRY_TYPE_NAME;
+import static io.prestosql.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static java.lang.String.format;
+
+/**
+ * A set of functions to convert between geometries and encoded polylines.
+ *
+ * @see <a href="https://developers.google.com/maps/documentation/utilities/polylinealgorithm">
+ *     https://developers.google.com/maps/documentation/utilities/polylinealgorithm</a> for a description of encoded polylines.
+ */
+public final class EncodedPolylineFunctions
+{
+    private static final Joiner OR_JOINER = Joiner.on(" or ");
+
+    private EncodedPolylineFunctions() {}
+
+    @Description("Decodes a polyline to a linestring")
+    @ScalarFunction("from_encoded_polyline")
+    @SqlType(GEOMETRY_TYPE_NAME)
+    public static Slice fromEncodedPolyline(@SqlType(StandardTypes.VARCHAR) Slice input)
+    {
+        return serialize(decodePolyline(input.toStringUtf8()));
+    }
+
+    private static OGCLineString decodePolyline(String polyline)
+    {
+        MultiPath multipath = new Polyline();
+        boolean isFirstPoint = true;
+
+        int index = 0;
+        int lat = 0;
```

Comment:
Rename to `latitude` / `longitude`. They are easier to read and follow that way.